packages/node_modules/pouchdb-selector-core/src/utils.js

```diff
@@ -308,22 +307,41 @@ function massageSelector(input) {
 
     if (typeof matcher !== 'object' || matcher === null) {
       matcher = {$eq: matcher};
-    } else if (!wasAnded) {
-      // These values must be placed in an array because these operators can be used multiple times on the same field
-      // when $and is used, mergeAndedSelectors takes care of putting them into arrays, otherwise it's done here:
-      if ('$ne' in matcher) {
-        matcher.$ne = [matcher.$ne];
-      }
-      if ('$regex' in matcher) {
-        matcher.$regex = [matcher.$regex];
-      }
     }
     result[field] = matcher;
   }
 
+  normalizeArrayOperators(result);
+
   return result;
 }
 
+//
+// The $ne and $regex values must be placed in an array because these operators can be used multiple times on the same field.
+// When $and is used, mergeAndedSelectors takes care of putting some of them into arrays, otherwise it's done here.
+//
+function normalizeArrayOperators(selector) {
+  var fields = Object.keys(selector);
+  for (var i = 0; i < fields.length; i++) {
+    var field = fields[i];
```

Comment:
It'd be better to use a forEach loop here...

```suggestion
  Object.keys(selector).forEach(function (field) {
    ...
  });
```

Reply:
Agreed! The forEach is more of my style, I must have been trying to replicate code style from above.